sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -55,7 +55,7 @@ trait ExpressionEvalHelper {
     val actual = try evaluate(expression, inputRow) catch {
       case e: Exception => fail(s"Exception evaluating $expression", e)
     }
-    if (actual != expected) {
+    if (actual !== CatalystTypeConverters.convertToCatalyst(expected)) {
```

Comment:
We used to use internal type for expressions tests, so `CatalystTypeConverters.convertToCatalyst` is not needed.

If you really think it's more convenient to use public types, please also change others test cases, to make them consistent.